src/app/services/users.service.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders, HttpParams } from '@angular/common/http';
+import { Observable, throwError } from 'rxjs';
+import { catchError, retry } from 'rxjs/operators';
+import { LoginModel } from 'src/app/login/login-model';
+@Injectable({ providedIn: 'root' })
+export class UsersService {
+    constructor(private http: HttpClient) { }
+    configUrl = "http://localhost:8080/login"
+
+    options: {
+        headers?: HttpHeaders | { [header: string]: string | string[]; };
+        observe?: 'body' | 'events' | 'response';
+        params?: HttpParams | { [param: string]: string | number | boolean | ReadonlyArray<string | number | boolean>; };
+        reportProgress?: boolean;
+        responseType?: 'arraybuffer' | 'blob' | 'json' | 'text';
+        withCredentials?: boolean;
+    } | undefined
+
+    httpOptions = {
+        headers: new HttpHeaders({
+            'Content-Type': 'application/x-www-form-urlencoded',
+            'Access-Control-Allow-Origin': '*',
+            'Access-Control-Allow-Methods': 'GET, POST, PATCH, PUT, DELETE, OPTIONS',
+            'Access-Control-Allow-Headers': 'Origin, Content-Type, X-Auth-Token'
+        })
+    };
```

Comment:
Let's move this common piece into separate common service, so same can be used by other service classes.